### PR TITLE
RFC: Use MADRs to records architecture, design and code decisions

### PR DESCRIPTION
**Pull request by irq0**
Uses template from https://adr.github.io/madr/

### CODE REVIEW

**Comment by jhmarina** [comment]
The linter is finding issues with other md files. I'm going to create an issue to sort them out separately.

I like the idea of using MADRs for architecture design & code. Other devs should have a look to, but it lgtm

**Comment by giubacc** [comment]
I think this process can be good in many ways, btw, I would like to highlight some thoughts:

1. what is a "decision" ? should we define what we intend with that? I constantly make decisions when developing/writing/designing, but I don't think every decision is worth to be reported to the entire team.
So if we give an approximation of what we are thinking when are we talking about a decision, that would be good, IMO.

2. everything that falls into "decision" category must therefore follow this process, is there a risk to add overhead?
3. could this process be addressed with regular meetings?
4. PR/issues aren't already defining a decision/feedback making process?

**Comment by jhmarina** [comment]
Currently, we have different approaches to discussing topics & coming up with decisions. We have Slack, calls & email. I try to get information from all mediums and summarize & document everything as good as I can. Which doesn't always work.

I think this approach could be seen as a voting mechanism. We discuss topics on different mediums but we resort to using ADR to reach a consensus.

I would like to give this approach a try and figure out if it works as a means of centralizing our decisions. We can try it for a few months and see how it goes.

Now, before going for it, it'd be nice to get feedback from other peers. @0xavi0, @jecluis, @m-ildefons, @votdev do you want to give it a try?

**Review by m-ildefons — APPROVED** [review]
I think this approach has a lot of benefits for an open source project.
The crux is and will always be that someone needs to take discussions that happened elsewhere (slack, emails, etc) and summarize them, but this would apply to other solutions like confluence as well, so I don't see that as a problem specific to this approach.
It's certainly a lot better than our current approach with slack. I quite like it.

**Review by jecluis — COMMENTED** [review]
lgtm

**Comment by jecluis** [comment]
Also, this will require a following PR to fix all the linting issues across the repo.

**Review by jecluis — APPROVED** [review]

**Comment by irq0** [comment]
> Also, this will require a following PR to fix all the linting issues across the repo.

@m-ildefons fixed the .md linter errors, thanks!

File: README.md [hunk 1]
```diff
@@ -22,74 +23,86 @@ applications.
...
+[here][3].
 
-## Docker
+### Podman
+
+```shell
+podman run --replace --name=s3gw -it -p 7480:7480 ghcr.io/aquarist-labs/s3gw:latest
 ```
+
+i### Docker
```
Comment (jecluis) [inline README.md:84]: this feels like a typo? That `i` in the beginning?

**Review by jecluis — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jecluis) [reply README.md:84]: It also breaks rendering the header.

Comment (irq0) [inline README.md:84]:
```suggestion
### Docker
```

Comment (irq0) [reply README.md]: Done

**Review by jecluis — APPROVED** [review]

**Pull request merged**